cogito/gchatsink_test.go

```diff
@@ -15,38 +15,62 @@ import (
 )
 
 func TestSinkGoogleChatSendSuccess(t *testing.T) {
-	wantGitRef := "deadbeef"
-	wantState := cogito.StateError // We want a state that is sent by default
-	var message googlechat.BasicMessage
-	reply := googlechat.MessageReply{}
-	var URL *url.URL
-	ts := testhelp.SpyHttpServer(&message, reply, &URL, http.StatusOK)
-	request := basePutRequest
-	request.Source.GChatWebHook = ts.URL
-	request.Params = cogito.PutParams{State: wantState}
-	request.Env = cogito.Environment{
-		BuildPipelineName: "the-test-pipeline",
-		BuildJobName:      "the-test-job",
+	type testCase struct {
+		name       string
+		setWebHook func(req *cogito.PutRequest, url string)
 	}
-	assert.NilError(t, request.Source.Validate())
-	sink := cogito.GoogleChatSink{
-		Log:     hclog.NewNullLogger(),
-		GitRef:  wantGitRef,
-		Request: request,
+
+	test := func(t *testing.T, tc testCase) {
+		wantGitRef := "deadbeef"
+		wantState := cogito.StateError // We want a state that is sent by default
+		var message googlechat.BasicMessage
+		reply := googlechat.MessageReply{}
+		var URL *url.URL
+		ts := testhelp.SpyHttpServer(&message, reply, &URL, http.StatusOK)
+		request := basePutRequest
+		request.Params = cogito.PutParams{State: wantState}
+		request.Env = cogito.Environment{
+			BuildPipelineName: "the-test-pipeline",
+			BuildJobName:      "the-test-job",
+		}
+		tc.setWebHook(&request, ts.URL)
```

Comment:
here we call the function, different per test case, to put in the struct the "webhook" from the test server.